devutils/git-hooks/pre-commit

```diff
@@ -44,33 +51,61 @@ PreCommit::BuildChecks () {
         go install github.com/bazelbuild/buildifier/buildifier
     fi
 
-    output=$("$GOPATH/bin/buildifier" -mode=check $filepath)
+    output=$("$GOPATH/bin/buildifier" -mode=check "$filepath")
     if [ -n "$output" ]; then
         Log::Warning "$output"
-        Log::Failure "formatting error in a BUILD file. Aborting..."
+        Log::Error "formatting error in a BUILD file."
+    fi
+}
+
+
+# Rules checking file is correctly registered in the commit
+PreCommit::GitChecks () {
+    local filepath="$1"
+
+    # Ensure commit contains the same file that the one currently in the project
+    if [ "$(git status --porcelain "$filepath" | cut -c2)" != ' ' ]; then
+        Log::Error "unstaged change detected, but file is staged in the commit"
     fi
 }
 
 
 main () {
-    cd $(bazel info workspace)
+    cd "$(bazel info workspace)" || Log::Failure "unable to go in the workspace root. Aborting..."
 
     # Ensures every tests affected by modified targets pass
     git diff --name-only --staged HEAD \
         | Bazel::GetAssociatedTests \
         | Bazel::TestTargets || Log::Failure "test failure, aborting commit."
```

Comment:
I think it's okay since it is *"a test failure"* and not a set of tests failing.